src/source/Ice/SocketConnection.c

```diff
@@ -132,8 +132,13 @@ STATUS socketConnectionInitSecureConnection(PSocketConnection pSocketConnection,
     ENTERS();
     TlsSessionCallbacks callbacks;
     STATUS retStatus = STATUS_SUCCESS;
+    BOOL locked = FALSE;
 
     CHK(pSocketConnection != NULL, STATUS_NULL_ARG);
+
+    MUTEX_LOCK(pSocketConnection->lock);
```

Comment:
From my understanding, the race occurs when we're accessing `pSocketConnection->localSocket`. I think the actual race happens in the outbound packet callback:

https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/73a6ec9ce5add1aab02bbda4cc31750739cffe7f/src/source/Ice/SocketConnection.c#L89-L100

`socketSendDataWithRetry` expects the caller to hold the socket connection lock.

Reply:
The particular TSAN issue was flagged detecting a race in connect and recvfrom API calls. We need to look closely on how we can better protect the socketConnectionIsConnected API against the ongoing socket operations. It's not trivial with the existing code form what I see.